example/spring-data-mongodb/src/main/kotlin/com/github/inflab/example/spring/data/mongodb/repository/ReplaceRootRepository.kt

```diff
@@ -0,0 +1,111 @@
+                    // TODO: add $mergeObjects expressions
+                    MergeObjects.merge(
+                        mapOf(
+                            "dogs" to 0,
+                            "cats" to 0,
+                            "birds" to 0,
+                            "fish" to 0,
+                        ),
+                    ).mergeWith("\$pets")
+                }
+            }
+        }
+
+        return mongoTemplate.aggregate(aggregation, PEOPLE, Pet::class.java)
+    }
+
+    /**
+     * @see <a href="https://www.mongodb.com/docs/manual/reference/operator/aggregation/replaceRoot/#-replaceroot-with-a-document-nested-in-an-array">ReplaceRoot with a Document Nested in an Array</a>
+     */
+    fun replaceRootWithNestedArray(): AggregationResults<Grade> {
+        val aggregation = aggregation {
+            unwind { path(Student::grades) }
+            match(Criteria.where((Student::grades..Grade::grade).toDotPath()).gte(90))
+            replaceRoot {
+                newRoot(Student::grades)
+            }
+        }
+
+        return mongoTemplate.aggregate(aggregation, STUDENT, Grade::class.java)
+    }
+
+    /**
+     * @see <a href="https://www.mongodb.com/docs/manual/reference/operator/aggregation/replaceRoot/#-replaceroot-with-a-newly-created-document">ReplaceRoot with a Newly Created Document</a>
+     */
+    fun replaceRootWithNewlyCreatedDocument(): AggregationResults<FindFullNameResult> {
+        val aggregation = aggregation {
+            replaceRoot {
+                newRoot {
+                    FindFullNameResult::fullName set {
+                        // TODO: add $concat expressions
+                        Concat.valueOf(Contact::firstName.toDotPath()).concat(" ").concatValueOf(Contact::lastName.toDotPath())
+                    }
+                }
+            }
+        }
+
+        return mongoTemplate.aggregate(aggregation, CONTACT, FindFullNameResult::class.java)
+    }
+
+    /**
+     * @see <a href="https://www.mongodb.com/docs/manual/reference/operator/aggregation/replaceRoot/#-replaceroot-with-a-new-document-created-from---root-and-a-default-document">ReplaceRoot with a New Document Created from $$ROOT and a Default Document</a>
+     */
+    fun replaceRootWithNewlyCreatedDocumentAndDefaultDocument(): AggregationResults<ContactInfo> {
+        val aggregation = aggregation {
+            replaceRoot {
+                expressions {
+                    // TODO: add $mergeObjects expressions
+                    MergeObjects.merge(
+                        mapOf(
+                            ContactInfo::id.toDotPath() to "",
+                            ContactInfo::name.toDotPath() to "",
+                            ContactInfo::email.toDotPath() to "",
+                            ContactInfo::cell.toDotPath() to "",
+                            ContactInfo::home.toDotPath() to "",
+                        ),
+                    ).mergeWith("\$\$ROOT")
```

Comment:
I think it would be good if aggregation variable is added later.

Reply:
`AggregationVariable` interface is added in spring mongodb 4.1.3.
so, we can use `SystemVariable.ROOT` because example module use latest version of spring data mongodb